pkg/util/globalerror/grpc_test.go

```diff
@@ -29,70 +29,124 @@ import (
-				expectedGrpcCode:   codes.DeadlineExceeded,
-				expectedContextErr: context.DeadlineExceeded,
+				ctxCanceled:         false,
+				ctxDeadlineExceeded: true,
+				origErr:             status.Error(codes.DeadlineExceeded, context.DeadlineExceeded.Error()),
+				expectedGrpcCode:    codes.DeadlineExceeded,
+				expectedContextErr:  context.DeadlineExceeded,
 			},
 			"gRPC DeadlineExceeded error": {
-				origErr:            grpcstatus.Error(codes.DeadlineExceeded, context.DeadlineExceeded.Error()),
-				expectedGrpcCode:   codes.DeadlineExceeded,
-				expectedContextErr: context.DeadlineExceeded,
+				ctxCanceled:         false,
+				ctxDeadlineExceeded: true,
+				origErr:             grpcstatus.Error(codes.DeadlineExceeded, context.DeadlineExceeded.Error()),
+				expectedGrpcCode:    codes.DeadlineExceeded,
+				expectedContextErr:  context.DeadlineExceeded,
 			},
 			"wrapped gogo DeadlineExceeded error": {
-				origErr:            errors.Wrap(status.Error(codes.DeadlineExceeded, context.DeadlineExceeded.Error()), "custom message"),
-				expectedGrpcCode:   codes.DeadlineExceeded,
-				expectedContextErr: context.DeadlineExceeded,
+				ctxCanceled:         false,
+				ctxDeadlineExceeded: true,
+				origErr:             errors.Wrap(status.Error(codes.DeadlineExceeded, context.DeadlineExceeded.Error()), "custom message"),
+				expectedGrpcCode:    codes.DeadlineExceeded,
+				expectedContextErr:  context.DeadlineExceeded,
 			},
 			"wrapped gRPC DeadlineExceeded error": {
-				origErr:            errors.Wrap(grpcstatus.Error(codes.DeadlineExceeded, context.DeadlineExceeded.Error()), "custom message"),
-				expectedGrpcCode:   codes.DeadlineExceeded,
-				expectedContextErr: context.DeadlineExceeded,
+				ctxCanceled:         false,
+				ctxDeadlineExceeded: true,
+				origErr:             errors.Wrap(grpcstatus.Error(codes.DeadlineExceeded, context.DeadlineExceeded.Error()), "custom message"),
+				expectedGrpcCode:    codes.DeadlineExceeded,
+				expectedContextErr:  context.DeadlineExceeded,
 			},
-			"ErrorWithStatus with Canceled status": {
-				origErr:            WrapErrorWithGRPCStatus(errors.New("cancel error"), codes.Canceled, nil),
-				expectedGrpcCode:   codes.Canceled,
-				expectedContextErr: context.Canceled,
+			"ErrorWithStatus with Canceled status and with canceled context": {
+				ctxCanceled:         true,
+				ctxDeadlineExceeded: false,
+				origErr:             WrapErrorWithGRPCStatus(errors.New("cancel error"), codes.Canceled, nil),
+				expectedGrpcCode:    codes.Canceled,
+				expectedContextErr:  context.Canceled,
+			},
+			"ErrorWithStatus with Canceled status and without canceled context": {
+				ctxCanceled:         false,
+				ctxDeadlineExceeded: false,
+				origErr:             WrapErrorWithGRPCStatus(errors.New("cancel error"), codes.Canceled, nil),
+				expectedGrpcCode:    codes.Canceled,
+				expectedContextErr:  nil,
 			},
```

Comment:
I'm afraid of the fact that if someone passes a wrong context to `WrapGRPCErrorWithContextError` it will never map the `context.Canceled` anymore, but I guess that's exactly what we want here.